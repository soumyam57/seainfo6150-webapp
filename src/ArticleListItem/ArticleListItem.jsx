import React from "react";

const ArticleListItem = props => {
  return (
    <div>
      <h3>{props.article.title}</h3>
      <time dateTime={props.article.timeStamp}>{props.article.displayDate}</time>
      <p>{props.article.shortText}</p>
    </div>
  );
};

export default ArticleListItem;