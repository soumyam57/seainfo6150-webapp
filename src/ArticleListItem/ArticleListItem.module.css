.container {
    background-image: linear-gradient(
        to right,
        #ff9999,
        white 30%
    );
    border: 1px solid #e60000;
    border-radius: 10px;
    margin-bottom: 20px;
    padding: 10px;
    position: relative;
    box-shadow: 0 0 5px 5px #8c8c8c;
    font-family:"Arial";
    font-size: smaller;
}

.title {
    font-family:"Arial";
    color: #e60000;
    font-size: 1.5rem;
    margin: .5rem 0;
}

.pubDate {
    font-family:"Arial";
    font-weight: bold;
    font-size: smaller;
}