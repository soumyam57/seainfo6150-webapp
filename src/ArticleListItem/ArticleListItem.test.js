import React from 'react';
import renderer from 'react-test-renderer';
import ArticleListItem from './ArticleListItem';

const article={
        author: "Kate Farley",
        displayDate: "November 22nd 2018, 7:12 am",
        shortText: "The Statue of Liberty's original torch is getting a new home",
        title: "The Statue of Liberty's torch heads to new museum"
}

describe("ArticlelistItem tests", () => {
    it("renders correctly when passed an object", () => {
    const { container } = render(<ArticleListItem article={article}/>);
      expect(container).toMatchSnapshot();
    });
});