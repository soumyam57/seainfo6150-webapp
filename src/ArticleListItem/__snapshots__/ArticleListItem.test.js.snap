// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlelistItem tests renders correctly when passed an object 1`] = `
<div>
  <li
    class="container"
  >
    <article
      class="article"
    >
      <div
        class="wrapper"
      >
        <h2
          class="title"
        >
          The Statue of Liberty's torch heads to new museum
        </h2>
      </div>
      <button
        class="button"
      >
        Show more
      </button>
    </article>
  </li>
</div>
`;
